src/Umbraco.Web.BackOffice/Controllers/ImagesController.cs

```diff
@@ -54,12 +54,20 @@ public IActionResult GetBigThumbnail(string originalImagePath)
         public IActionResult GetResized(string imagePath, int width)
         {
             var ext = Path.GetExtension(imagePath);
-
+            
+            // check if imagePath is local to prevent open redirect
+            if (!Uri.IsWellFormedUriString(imagePath, UriKind.Relative))
```

Comment:
Hi @mbentancour, I think an additional check on leading `//` may be required for protocol-relative urls, such as `//umbraco.com/test.png`. This URI passes the `Uri.IsWellFormedUriString(imagePath, UriKind.Relative)` test, but it would still redirect the user to a different domain.

Reply:
Nice catch. I added an extra check for isLocalUrl(imageUrl), and then replaced RedirectResult with LocalRedirectResult. 
I tested it with and without escaped characters, with and without "https://..." and it seems to do the trick.

Feel free to give it another try.

Reply:
I'll mark this conversation as resolved since it seems to work fine, but if you manage to get around this fix let me know and I'll give it another look :)

Reply:
I've tested the branch locally and it seems to be working as expected, replying 401 to any non-relative paths 👍